src/utils/treeUtils.ts

```diff
@@ -15,4 +15,18 @@ export namespace treeUtils {
     function getResourcesUri(): Uri {
         return Uri.joinPath(ext.context.extensionUri, 'resources')
     }
+
+    export function findNearestParent<T extends AzExtTreeItem>(node: AzExtTreeItem, parent: T): T | null {
+        const parentInstance: string = parent.constructor.name;
+        let foundParent: boolean = false;
+        let currentNode: AzExtTreeItem = node;
+        while (currentNode.parent) {
+            if (currentNode.constructor.name === parentInstance) {
+                foundParent = true;
+                break;
+            }
+            currentNode = currentNode.parent;
+        }
+        return foundParent ? currentNode as T : null;
```

Comment:
I think you should throw some sort of error here if you don't get a parent. If it returns as `null`, you'll get some funky errors in other parts of code. That way you won't have to do any `as T` typings either.

Reply:
Good idea, this looks way cleaner now